website/pages/docs/reference/source-spec.md

```diff
@@ -142,3 +126,28 @@ If set to `true`, the exporter will not verify the server will connect via `http
 (`object`, optional)
 
 Plugin-specific configurations. Visit [source plugins](/docs/plugins/sources/overview) documentation for more information.
+The following options are available for most source plugins:
+
+#### concurrency
+
+(`int`, optional, default: `<unique-per-plugin>`, introduced in CLI `v3.6.0`)
+
+A best effort maximum number of Go routines to use. Lower this number to reduce memory usage.
+
+#### scheduler
+
+(`string`, optional, default: `dfs`, introduced in CLI `v3.6.0`, **EXPERIMENTAL**)
+
+The scheduler to use when determining the priority of resources to sync. Currently, the only supported values are `dfs` (depth-first search) and `round-robin`. This is an experimental feature, and may be removed in the future. For more information about this, see [performance tuning](/docs/advanced-topics/performance-tuning).
+
+#### backend
+
+(`string`, optional, default: `local`, introduced in CLI `v3.6.0`)
+
+The backend to use for storing the state of incremental tables. Currently, the only supported value is `local` (store the state in the local filesystem). For more information, see [managing incremental tables](/docs/advanced-topics/managing-incremental-tables).
+
+#### backend_spec
```

Comment:
`backend_spec` is deprecated since v4, it's now called `backend_options`, and contains `table_name` and `connection`. It supports referencing values from other plugins, as documented in the blog post: https://www.cloudquery.io/blog/proto-v3 

Would be great to update the documentation for this here